### DISCUSSION TITLE
Handling events with short artificial gaps

### DISCUSSION THREAD

**Discussion by olemp** [Puzzlepart/did#868] [Ideas]
### TLDR
Recently it's become common to end meetings 10 minutes before the hour, e.g. have a meeting run from 0900 - 0950. This should be logged as a 1 hour meeting. The gap is to allow people go to the restroom etc before any subsequent events. 

![image](https://user-images.githubusercontent.com/170147/109017127-2d895600-76b7-11eb-9570-54f5a50dbccf.png)

### Discussion
Currently Did does not give any visual cues that there is a gap between events. This can potentially cause you to log fewer hours, especially if you have many meetings with these characteristics

Additionally we do not have a consistent story on how to fix this in did. The premise for Did is that the calendar should be the timesheet, i.e. all logged hours in Did are equivalent to the appointments you have in your calendar. Adding functionality in Did where we allow did to count a 50 minute meeting as 1 hour can be confusing and may break the expectation that your calendar and Did is 1:1, at least when it comes to non-ignored events. 

My current workaround is to 
1) extend the any such events using the outlook app so that it ends on the hour
2) ignore the event and create a new event in my calendar which correctly corresponds to the hours which will be billed

Any suggestions or ideas on possible ways to improve the experience?

### UI implementation suggestion
* Given events `A` and `B`
* The difference in `A.endtime` and `B.startTime` being >15 min
* Display the gap in minutes in the timesheet in a similar fashion to this

![image](https://user-images.githubusercontent.com/7300548/109017709-b99b7d80-76b7-11eb-80d4-c2c40f9d81db.png)

#### Dummy gridCell markup
```
<div class="verticalLine" style="
    height: 50px;
    width: 1px;
    border-left: 1px dotted #000;
    position: absolute;
"> </div>
    <div class="freetime" style="
    z-index: 2;
    background: #f3f2f1;
    padding: 1px 10px;
">30min</div>
```

**Comment by olemp** [comment]
Not sure about this one. Is it worth it? cc @damsleth @okms

**Reply by damsleth** [reply]
It's my personal opinion that did and the calendar should be 1:1, _always_. 
Meaning, did should never fudge time entries coming from the calendar one way or the other.  

What I _think_ did should do is highlight gaps in the calendar, like in the proposed UI implementation above.  
The only downside to this approach is that ignored events will display as gaps in the calendar, and be potentially confusing.  
Still, I think the upside of visualising gaps in the calendar is worth doing.

Down the line I think we could consider having the option of displaying ignored events as well, in which case they would be highlighted as "not included in the timesheet" somehow.

**Reply by olemp** [reply]
It could be a user setting or something - "show time gaps" or whatever. For me it's enough to see that I have full days (8 hours).

Displaying ignored events in the Timesheet is another discussion. It can "disturb" the Timesheet, but maybe an event was ignored "by accident".

**Reply by olemp** [reply]
I started on a [PR](https://github.com/Puzzlepart/did/pull/881). You could look at the styling and markup in the [VisualGap](https://github.com/Puzzlepart/did/blob/feature/artificial-gaps-868/client/components/EventList/TimeColumn/VisualGap.tsx) component.

**Comment by olemp** [comment]
This is implemented.
